src/mam4xx/mo_gas_phase_chemdr.hpp

```diff
@@ -425,12 +425,12 @@ void perform_atmospheric_chemistry_and_microphysics(
   for (int m = 0; m < num_modes; ++m) {
     const int ll = config_setsox.lptr_so4_cw_amode[m] - offset_aerosol;
     if (0 <= ll) {
-      const auto aqso4 = ekat::subview(dqdt_aqso4, ll);
+      const auto aqso4   = ekat::subview(dqdt_aqso4, ll);
       const auto aqh2so4 = ekat::subview(dqdt_aqh2so4, ll);
-      aqso4_flx[m] = aero_model::calc_sfc_flux(team, aqso4, pdel, nlev);
-      aqh2so4_flx[m] = aero_model::calc_sfc_flux(team, aqh2so4, pdel, nlev);
-      // outfld( trim(cnst_name_cw(mm))//'AQSO4', sflx(:ncol), ncol, lchnk)
-      // outfld( trim(cnst_name_cw(mm))//'AQH2SO4', sflx(:ncol), ncol, lchnk)
+      const Real vmr_so4 = aero_model::calc_sfc_flux(team, aqso4, pdel, nlev);
+      const Real vmr_h2s = aero_model::calc_sfc_flux(team, aqh2so4, pdel, nlev);
+      aqso4_flx[m]       = conversions::mmr_from_vmr(vmr_so4, adv_mass_kg_per_moles[ll]);
+      aqh2so4_flx[m]     = conversions::mmr_from_vmr(vmr_h2s, adv_mass_kg_per_moles[ll]); 
```

Comment:
Thanks @overfelt for fixing this. @kaizhangpnl : In F90 codes, this factor (`conversions::mmr_from_vmr`, see function definition [here](https://github.com/eagles-project/mam4xx/blob/3baae75c4f5819c0f34c29a3007d078caf54a3f9/src/mam4xx/conversions.hpp#L58)) is applied before calling `calc_sfc_flux` (see F90 code [here](https://github.com/eagles-project/e3sm_mam4_refactor/blob/2e58607847ef2868e390c564039e8f04d648d9ac/components/eam/src/chemistry/modal_aero/sox_cldaero_mod.F90#L301)).

They seem equivalent to me mathematically, as the factor is a constant for a particular species, but I just wanted to run this by you in case I am missing something.